src/interface_r/R/type_resolvers.R

```diff
@@ -0,0 +1,18 @@
+as_nullable_integer <- function(x) {
+  if (is.null(x))
+    x
+  else
+    as.integer(x)
+}
+
+resolve_model_input <- function(x) {
+  if (is.matrix(x)) {
+    x
+  } else if (is.data.frame(x)) {
```

Comment:
This should cover `data.table`, as well. But good to double check.

Reply:
This is okay; `is.data.frame(x)` returns TRUE when passed a data.table or tibble as well.